### PR TITLE
[core] Deprecate QuotaAwareEvictionPolicy

### PR DESCRIPTION
**Pull request by scv119**
QuotaAwareEvictionPolicy is no longer used. This PR deprecates it. It will help the ongoing plasmastore refactor too.

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

File: src/ray/object_manager/plasma/store.cc [hunk 1]
```diff
@@ -924,14 +913,8 @@ Status PlasmaStore::ProcessMessage(const std::shared_ptr<Client> &client,
...
   case fb::MessageType::PlasmaSetOptionsRequest: {
-    std::string client_name;
-    int64_t output_memory_quota;
-    RAY_RETURN_NOT_OK(
-        ReadSetOptionsRequest(input, input_size, &client_name, &output_memory_quota));
-    client->name = client_name;
-    bool success = eviction_policy_.SetClientQuota(client.get(), output_memory_quota);
-    RAY_RETURN_NOT_OK(SendSetOptionsReply(
-        client, success ? PlasmaError::OK : PlasmaError::OutOfMemory));
+    RAY_LOG(WARNING) << "Set client quota deprecated.";
```
Comment (rkooo567) [inline src/ray/object_manager/plasma/store.cc:916]: Maybe removing the whole code path for this?

Comment (scv119) [reply src/ray/object_manager/plasma/store.cc:916]: image
yes sir.

File: python/ray/worker.py [hunk 2]
```diff
@@ -858,7 +855,6 @@ def init(
         mode=driver_mode,
         log_to_driver=log_to_driver,
         worker=global_worker,
-        driver_object_store_memory=_driver_object_store_memory,
```
Comment (scv119) [inline python/ray/worker.py:861]: this changes public API, should we leave it there and add some warning instead?

Comment (ericl) [reply python/ray/worker.py:861]: Yea probably should log a warning that it will be removed.

**Review by ericl — APPROVED** [review]

**Review by suquark — APPROVED** [review]

**Comment by scv119** [comment]
add warning on using driver_object_store_memory.

**Comment by scv119** [comment]
```
>>> ray.init(_driver_object_store_memory=100 * 1024 * 1024)
2021-07-07 00:33:58,300	WARNING worker.py:1292 -- `driver_object_store_memory` is deprecated and will be removed in the future.
{'node_ip_address': '192.168.86.195', 'raylet_ip_address': '192.168.86.195', 'redis_address': '192.168.86.195:6379', 'object_store_address': '/tmp/ray/session_2021-07-07_00-33-55_753527_27860/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2021-07-07_00-33-55_753527_27860/sockets/raylet', 'webui_url': None, 'session_dir': '/tmp/ray/session_2021-07-07_00-33-55_753527_27860', 'metrics_export_port': 62697, 'node_id': 'b2251cc722a3e967a6b0911a1353ef52501eddd5be77497a96b223c1'}
```

**Comment by scv119** [comment]
confirmed that the failed test (//python/ray/serve:test_deploy) is broken on base commit.

**Pull request merged**